### PR TITLE
Add YamlProvider.order_mode natural is default, adds simple

### PR DESCRIPTION
**Pull request by ross**
Natural (existing default):

```yaml
  config:
    class: octodns.provider.yaml.YamlProvider
    directory: ./config
    enforce_order: True
    order_mode: natural
```

```yaml
'*.1.2': 'a'
'*.2.2': 'b'
'*.10.1': 'c'
'*.11.2': 'd'
```

Simple (new option):

```yaml
  config:
    class: octodns.provider.yaml.YamlProvider
    directory: ./config
    enforce_order: True
    order_mode: simple
```

```yaml
'*.1.2': 'a'
'*.10.1': 'c'
'*.11.2': 'd'
'*.2.2': 'b'
```

/cc Fixes https://github.com/octodns/octodns/issues/1206 
/cc @stantona

### CODE REVIEW

File: octodns/provider/yaml.py [hunk 1]
```diff
@@ -34,6 +34,11 @@ class YamlProvider(BaseProvider):
         # Whether or not to enforce sorting order when loading yaml
         # (optional, default True)
         enforce_order: true
+        # What sort mode to employ when enforcing order
+        # - simple: `sort`
+        # - natual: https://pypi.org/project/natsort/
+        # (optional, default natual)
```
Comment (stantona) [inline octodns/provider/yaml.py:40]: Small typo: `natual` -> `natural`.

**Review by stantona — APPROVED** [review]

**Comment by ross** [comment]
Went ahead and added support for dump so that `YamlProvider` can write things out in both natural and simple order. Now has error handling when an unrecognized order mode is specified.

**Review by ross — COMMENTED** [review]
/ci modules bump

**Review by ross — COMMENTED** [review]
/ci modules bump

**Review by stantona — APPROVED** [review]

**Pull request merged**